flocker/node/test/test_script.py

```diff
@@ -22,37 +23,52 @@
 from ..script import (
     AgentScript, ContainerAgentOptions,
     AgentServiceFactory, DatasetAgentOptions, validate_configuration,
-    _context_factory_and_credential, GenericAgentScript,
+    _context_factory_and_credential, DatasetServiceFactory,
+    AgentService, BackendDescription, get_configuration,
+    DeployerType,
 )
 
 from .._loop import AgentLoopService
-from .._deploy import P2PManifestationDeployer
-from ...testtools import MemoryCoreReactor
+from ...testtools import MemoryCoreReactor, random_name
 from ...ca.testtools import get_credential_sets
 
 
-def setup_config(test):
+def setup_config(test, control_address=u"10.0.0.1", control_port=1234,
+                 name=None):
     """
     Create a configuration file and certificates for a dataset agent in a
     temporary directory.
 
-    Sets ``config`` attribute on the test instance with the path to the
-    config file.
+    Sets ``config`` attribute on the test instance with the path to the config
+    file.
 
     :param test: A ``TestCase`` instance.
+    :param unicode control_address: The address of the control service.
+    :param int control_port: The port number of the control service.
+    :param unicode name: The ZFS pool name.  If ``None``, a random one will be
+        generated that identifies the given test.
     """
-    ca_set, _ = get_credential_sets()
+    if name is None:
+        name = random_name(test)
+    ca_set = get_credential_sets()[0]
     scratch_directory = FilePath(test.mktemp())
     scratch_directory.makedirs()
     test.config = scratch_directory.child('dataset-config.yml')
     test.config.setContent(
         yaml.safe_dump({
             u"control-service": {
-                u"hostname": u"10.0.0.1",
-                u"port": 1234,
+                u"hostname": control_address,
+                u"port": control_port,
             },
             u"dataset": {
                 u"backend": u"zfs",
+                u"name": name,
+                # XXX Can't put FilePath in a yaml file (well you can but
```

Comment:
XXX to me implies that there is work to be done, but it doesn't seem like there is so remove.